File: server/scripts/addCredits.js
import "dotenv/config";
import { addCredits, getUserByUsername, initDb } from "../db.js";

const parseArgs = () => {
  const args = process.argv.slice(2);
  const map = {};
  for (let i = 0; i < args.length; i += 1) {
    const key = args[i];
    if (!key.startsWith("--")) continue;
    map[key.replace("--", "")] = args[i + 1];
    i += 1;
  }
  return map;
};

const main = async () => {
  const args = parseArgs();
  const username = args.username;
  const delta = Number.parseInt(args.delta ?? "0", 10);
  const reason = args.reason ?? "ajuste_manual";
  const createdBy = args.createdBy ?? null;

  if (!username || !delta) {
    console.error("Usage: node server/scripts/addCredits.js --username <email> --delta <valor> [--reason texto]");
    process.exit(1);
  }

  await initDb();
  const user = await getUserByUsername(username);
  if (!user) {
    console.error("User not found");
    process.exit(1);
  }

  const credits = await addCredits({
    userId: user.id,
    delta,
    reason,
    createdBy,
  });

  console.log("Credits updated:");
  console.log(`  user_id: ${user.id}`);
  console.log(`  username: ${user.username}`);
  console.log(`  credits: ${credits}`);
};

main().catch((error) => {
  console.error(error.message || error);
  process.exit(1);
});

File: backend/scripts/add_credits.py
from __future__ import annotations

import argparse
from pathlib import Path
import sys

BASE_DIR = Path(__file__).resolve().parents[1]
sys.path.append(str(BASE_DIR))

from app.config import settings  # noqa: E402
from app.db import add_credits, get_user_by_username, init_db  # noqa: E402


def main() -> None:
    parser = argparse.ArgumentParser(description="Add credits to a user")
    parser.add_argument("--username", required=True)
    parser.add_argument("--delta", type=int, required=True)
    parser.add_argument("--reason", default="ajuste_manual")
    parser.add_argument("--created-by", default=None)
    parser.add_argument(
        "--db-path",
        default=str(settings.db_path),
        help="Path to SQLite database file",
    )

    args = parser.parse_args()
    db_path = Path(args.db_path)

    init_db(db_path)
    user = get_user_by_username(db_path, args.username)
    if not user:
        raise SystemExit("User not found")

    new_total = add_credits(db_path, user.id, args.delta, args.reason, args.created_by)

    print("Credits updated:")
    print(f"  user_id: {user.id}")
    print(f"  username: {user.username}")
    print(f"  credits: {new_total}")


if __name__ == "__main__":
    main()
